### ISSUE TITLE
NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' 

### ISSUE THREAD

**Issue by jemshit** [OneSignal/OneSignal-Android-SDK#729]
**Description:**

Nullpointer exceptions on some users:
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
       at org.json.JSONTokener.nextCleanInternal(JSONTokener.java:116)
       at org.json.JSONTokener.nextValue(JSONTokener.java:94)
       at org.json.JSONObject.(JSONObject.java:156)
       at org.json.JSONObject.(JSONObject.java:173)
       at com.onesignal.OneSignalRemoteParams.processJson(OneSignalRemoteParams.java:74)
       at com.onesignal.OneSignalRemoteParams.access$000(OneSignalRemoteParams.java:11)
       at com.onesignal.OneSignalRemoteParams$1.onSuccess(OneSignalRemoteParams.java:58)
       at com.onesignal.OneSignalRestClient$5.run(OneSignalRestClient.java:253)
       at java.lang.Thread.run(Thread.java:761)
```
[This is the line](https://github.com/OneSignal/OneSignal-Android-SDK/blob/b59d4562a6f1fe9d4bf85a66636469c39dfa9822/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalRemoteParams.java#L74), the code catches `JSONException` but `NullpointerException` is thrown.

**Environment**

1. Android SDK 28
2. Onesignal latest (3.10.7)
3. Added onesignal to project using gradle as described in documentation

**Steps to Reproduce Issue:**

1. SDK is added
2. Push notifications are received, tags are sent
3. NullpointerException occurs on some users, so can't reproduce. But code is available for such Exception flow, it can be avoided

**Comment by kingfisherphuoc** [comment]
I got this issue also.

**Comment by jkasten2** [comment]
@jemshit @kingfisherphuoc Thanks for reporting, what volume of crashes are you seeing with this error? And it is specific to any Android versions or device models?

We are looking into root cause of this exception if possible, if not I believe catching `Throwable` and adding a retry for the network request will be the next best course.

**Comment by kingfisherphuoc** [comment]
@jkasten2 I am facing this issue on all android versions. It occurs mostly on Android 6.0 as the image below:
<img width="1148" alt="Screen Shot 2019-04-10 at 9 20 47 AM" src="https://user-images.githubusercontent.com/962484/55847032-0e8c8400-5b72-11e9-8d76-9d35b57dbf7e.png">

**Comment by jemshit** [comment]
49% General Mobile phones
62% Android 7
Small amount of users

![image](https://user-images.githubusercontent.com/1948828/55878885-0e0de080-5ba6-11e9-9f13-b946ccc57e1e.png)

**Comment by gitton** [comment]
got the same issue.

**Comment by tuinui** [comment]
got same issue

**Comment by 9468lgy** [comment]
got the same issue.

**Comment by jkasten2** [comment]
Thanks for confirming the issue. It is now fixed in the [3.10.8 release](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.10.8)
